dist/gmusic.js

```diff
@@ -207,6 +248,15 @@ proto.playback = {
       return GMusic.Playback.NO_SHUFFLE;
     }
   },
+  setShuffle: function (mode) {
+    if (mode === GMusic.Playback.NO_SHUFFLE || mode === GMusic.Playback.ALL_SHUFFLE) {
+      while (this.playback.getShuffle() !== mode) {
+        this.playback.toggleShuffle();
+      }
+    } else {
+      console.warn('Invalid shuffle mode supplied');
```

Comment:
As a bonus, we can be nicer to developers by stating the value we received and expect:

```js
assert(GMusic.Playback.SHUFFLE_MODES.indexOf(mode) !== -1,
    'Expected shuffle mode "' + mode + '" to be inside ' + JSON.stringify(GMusic.Playback.SHUFFLE_MODES) + ' but it wasn\'t');
// while ...
```